fixed braziers for wintertodt + rs2grounditem.loot

ObserveObserver opened the pull request:
fixed braziers for wintertodt, fixed bug in wintertodt where player would be stuck in WAITING state (i think i did, anyway), fixed rs2grounditem.loot for 50 range instead of 255. tutorialisland fix still.

Commit:
ui now uses sprites to find interface elements. stopgap measure for sure. pathing no longer gets stuck at rat gates or western anvil etc. also added small pauses as antiban/more human-like inputs.

ObserveObserver committed:
removed debug stuff

oops

ObserveObserver committed:
Merge branch 'chsami:main' into main

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java:
```diff
@@ -363,14 +377,14 @@ private boolean handleBankLogic(MWintertodtConfig config) {
+            Rs2Bank.withdrawX(true, "hammer", 1);
+        }
         if (!Rs2Equipment.hasEquipped(ItemID.BRUMA_TORCH)) {
-            Rs2Bank.withdrawX(true, "tinderbox", 1, true);
```
why did you remove the true statement here?

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java:
```diff
@@ -354,7 +368,7 @@ private boolean handleBankLogic(MWintertodtConfig config) {
         if (Rs2Player.getWorldLocation().distanceTo(bankLocation) > 6) {
             Rs2Walker.walkTo(bankLocation);
-            Rs2Player.waitForWalking();
+            sleepUntil(() -> Rs2GameObject.findChest() != null);
```
this is more resource intensive, not sure if this is really needed.

ObserveObserver replied:
it was an attempt to get to open crates after it saw the bank such that it loads into loot tracker plugin (as it only loads into it after you're out of wintertodt doors)

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java:
```diff
@@ -95,26 +94,28 @@ public boolean run(MWintertodtConfig config) {
                     if (!isWintertodtAlive) {
-                        if (state != State.ENTER_ROOM && state != State.WAITING && state != State.BANKING) {
+                        if ((state != State.ENTER_ROOM && state != State.WAITING && state != State.BANKING)
+                        || Rs2Inventory.hasItem("supply crate")) {
```
use SUPPLY_CRATE to be consistent

chsami commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/wintertodt/MWintertodtScript.java:
```diff
@@ -381,6 +395,7 @@ private boolean handleBankLogic(MWintertodtConfig config) {
         Rs2Bank.withdrawX(config.food().getId(), config.foodAmount() - foodCount);
-        return sleepUntilTrue(() -> Rs2Inventory.hasItemAmount(config.food().getName(), config.foodAmount(), false, true), 100, 5000);
+        sleepUntil(() -> Rs2Inventory.hasItemAmount(config.food().getName(), config.foodAmount(), false, true));
+        return false;
```
this is weird, why are you always returning false?

ObserveObserver replied:
i have no idea where that came from, maybe some check i deleted early on

ObserveObserver committed:
Merge branch 'chsami:main' into main

Commit:
fixed prayer + added release invokemenu action to rs2inventory

Commit:
hunter script beta initial release, in scala and can be very finnicky with lombok.

has salamanders + crimson butterflies. will be updated overtime.

pom updated to include scala compiler and scala options.

Commit:
Merge remote-tracking branch 'origin/main' into sezfork

ObserveObserver committed:
housekeeping changes
